stdlib/experimental/prometheus/from.go

```diff
@@ -0,0 +1,339 @@
+	Type      string                 `json:"string"` // need to add other funcs for validation; first make sure this is right
+}
+
+func (p *PrometheusIterator) Connect(ctx context.Context) error {
+
+	return nil
+}
+
+func (p *PrometheusIterator) parse(r io.Reader, header http.Header) (err error) {
+	var parser expfmt.TextParser
+	now := time.Now()
+
+	mediatype, params, err := mime.ParseMediaType(header.Get("Content-Type"))
+	if err != nil {
+		return err
+	} // why do we need to parse the media type?
+
+	// Prepare output
+	metricFamilies := make(map[string]*dto.MetricFamily)
+	if mediatype == "application/vnd.google.protobuf" &&
+		// read about protobufs
+		params["encoding"] == "delimited" &&
+		params["proto"] == "io.prometheus.client.MetricFamily" {
+		for {
+			mf := &dto.MetricFamily{}
+			if _, err := pbutil.ReadDelimited(r, mf); err != nil {
+				if err == io.EOF {
+					break
+				}
+				return fmt.Errorf("reading metric family protocol buffer failed: %s", err)
+			}
+			metricFamilies[mf.GetName()] = mf
+		}
+	} else {
+		metricFamilies, err = parser.TextToMetricFamilies(r)
+		if err != nil {
+			return fmt.Errorf("reading text format failed: %s", err)
+		}
+	}
+
+	p.metrics = make([]Metric, 0)
+
+	// read metrics
+	for name, family := range metricFamilies {
+		for _, m := range family.Metric {
+			// read tags
+			tags := makeLabels(m)
+
+			// read fields
+			fields := make(map[string]interface{})
+
+			switch family.GetType() {
+			case dto.MetricType_SUMMARY:
+				continue // skip for now
+			case dto.MetricType_HISTOGRAM:
+				continue // skip for now
+			default: // counters, gagues and uptyped
+				fields = getNameAndValue(m)
+			}
+
+			// convert to row for flux Table
+			if len(fields) > 0 {
+				tm := now
+				if m.TimestampMs != nil && *m.TimestampMs > 0 {
+					tm = time.Unix(0, *m.TimestampMs*1000000)
+				}
+				met := Metric{
+					Timestamp: tm,
+					Tags:      tags,
+					Fields:    fields,
+					Name:      name,
+					// Type:      string(family.GetType()),
+				}
+				p.metrics = append(p.metrics, met)
+			}
+
+		}
+	}
+
+	return nil
+}
+
+func (p *PrometheusIterator) Fetch(ctx context.Context) (bool, error) {
+	p.url = "http://demo.robustperception.io:9090/metrics"
+	resp, err := http.Get(p.url)
+	if err != nil {
+		return false, err
+	}
+	defer resp.Body.Close()
+
+	// parse the entire body into a slice of Metrics
+	err = p.parse(resp.Body, resp.Header)
+	if err != nil {
+		return false, err
+	}
+
+	// Iterate over all Metrics in the slice
+	if p.i < len(p.metrics) {
+		p.nextMetric()
+		return true, nil
+	}
+	// p.met = p.metrics[0]
+	return false, nil
+}
+
+func (p *PrometheusIterator) nextMetric() error {
+	fmt.Println(p.i)
+	next := p.metrics[p.i]
+	p.i++
+	p.met = next
+
+	return nil
+}
+
+func (p *PrometheusIterator) Decode(ctx context.Context) (table flux.Table, err error) {
+
+	// unpacking Fields
+	var fieldVal string
+	var val interface{}
+	for f, v := range p.met.Fields {
```

Comment:
the outcome of this loop is  fieldVal, val := p.met.Fields[len(p.met.Fields) - 1] .   Is that what's intended?  Are there multiple Fields to decode here?

Reply:
Yeah, I was trying to figure out how to unpack the map values when I don't have the key. This was a hack-y way to unpack those values.

Reply:
Since keys in this map are strings, I don't think we can use an int to help with unpacking it.